src/System.Private.CoreLib/Tools/GenUnicodeProp/Program.cs

```diff
@@ -0,0 +1,386 @@
+        {
+            if (Verbose)
+                Console.WriteLine($"[{value}]");
+
+            var values = value.Split(',');
+            var unicodeCategoryValue = values[0];
+            var bidiCategoryValue = values[1];
+
+            return new[] { GetUnicodeCategoryValue(unicodeCategoryValue), GetBiDiCategoryValue(bidiCategoryValue) };
+        }
+
+        private static byte[] GetNumericValueBytes(string value)
+        {
+            double d;
+            var i = value.IndexOf('/');
+            if (i < 0)
+                d = double.Parse(value, CultureInfo.InvariantCulture);
+            else
+                d = double.Parse(value.Substring(0, i), CultureInfo.InvariantCulture) / double.Parse(value.Substring(i + 1), CultureInfo.InvariantCulture);
+            return BitConverter.GetBytes(d);
+        }
+
+        private static byte[] GetDigitValueBytes(string value)
+        {
+            if (Verbose)
+                Console.WriteLine($"[{value}]");
+
+            var values = value.Split(',');
+            var decimalDigitValue = values[0];
+            var digitValue = values[1];
+
+            return new[] { byte.Parse(decimalDigitValue), byte.Parse(digitValue) };
+        }
+
+        /// <summary>
+        /// Map a Unicode category symbol in UnicodeData.txt to a numeric value
+        /// </summary>
+        /// <param name="str">A two-letter abbreviation of Unicode category</param>
+        /// <returns>A numeric value for the corresponding two-letter Unicode category</returns>
+        private static byte GetUnicodeCategoryValue(string str)
+        {
+            return UnicodeCategoryMap.TryGetValue(str, out var v) ? v : throw new ArgumentException($"The str [{str}] is not a valid two-letter Unicode category.");
+        }
+
+        private static byte GetBiDiCategoryValue(string str)
+        {
+            return BiDiCategory.TryGetValue(str, out var v) ? v : throw new ArgumentException($"The str [{str}] is not a valid BiDi category.");
+        }
+
+        private static void PrintSourceIndexArray(string tableName, DataTable d, StreamWriter file)
+        {
+            Console.WriteLine("    ******************************** .");
+
+            var levels = d.GetBytes();
+
+            PrintByteArray(tableName, file, levels[0]);
+            PrintByteArray(tableName.Replace('1', '2'), file, levels[1]);
+            PrintByteArray(tableName.Replace('1', '3'), file, levels[2]);
+        }
+
+        private static void PrintValueArray(string tableName, FlatDataTable d, StreamWriter file)
+        {
+            Console.WriteLine("    ******************************** .");
+            PrintByteArray(tableName, file, d.GetBytesFlat());
+        }
+
+        private static void PrintByteArray(string tableName, StreamWriter file, byte[] str)
+        {
+            file.Write("\n        private static ReadOnlySpan<byte> " + tableName + " => new byte[" + str.Length + "]\n        {\n");
```

Comment:
>(" [](start = 22, length = 2)

would using $".." would be better here?

Reply:
I think it was this way in the perl script also, doesn't make much difference either way.